app/src/main/java/org/oppia/app/profile/AddProfileActivityPresenter.kt

```diff
@@ -0,0 +1,178 @@
+package org.oppia.app.profile
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.provider.MediaStore
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.inputmethod.InputMethodManager
+import android.widget.ImageView
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.oppia.app.R
+import org.oppia.app.activity.ActivityScope
+import org.oppia.app.databinding.AddProfileActivityBinding
+import org.oppia.app.viewmodel.ViewModelProvider
+import org.oppia.domain.profile.ProfileManagementController
+import javax.inject.Inject
+
+const val GALLERY_INTENT_RESULT_CODE = 1
+
+/** The presenter for [AddProfileActivity]. */
+@ActivityScope
+class AddProfileActivityPresenter @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val profileManagementController: ProfileManagementController,
+  private val viewModelProvider: ViewModelProvider<AddProfileViewModel>
+) {
+  lateinit var uploadImageView: ImageView
+  private val addViewModel by lazy {
+    getAddProfileViewModel()
+  }
+  private var selectedImage: Uri? = null
+  var allowDownloadAccess = false
+  var inputtedPin = false
+  var inputtedConfirmPin = false
+
+  @ExperimentalCoroutinesApi
+  fun handleOnCreate() {
+    activity.title = "Add Profile"
+    activity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    activity.supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_close_white_24dp)
+
+    val binding = DataBindingUtil.setContentView<AddProfileActivityBinding>(activity, R.layout.add_profile_activity)
+
+    binding.apply {
+      viewModel = addViewModel
+    }
+
+    binding.allowDownloadSwitch.setOnCheckedChangeListener { _, isChecked ->
+      allowDownloadAccess = isChecked
+    }
+
+    binding.infoIcon.setOnClickListener {
+      showInfoDialog()
+    }
+
+    addTextChangeListeners(binding)
+
+    binding.uploadImageButton.setOnClickListener {
+      val galleryIntent = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI)
+      activity.startActivityForResult(galleryIntent, GALLERY_INTENT_RESULT_CODE)
+    }
+    uploadImageView = binding.uploadImageButton
+
+    binding.createButton.setOnClickListener {
```

Comment:
I suggest breaking this up into several constituent helper methods to aid with readability.